### PR TITLE
Create the mentee mentor card

### LINKED ISSUE #9: Create manage mentees view
**Is your feature request related to a problem? Please describe.**
- Need to develop the view to manage mentees on the admin dashboard.

**Describe the solution you'd like**
- Create a dummy UI adhering to the given view on the admin wireframes.

**Additional context**
- This is the view that needs to be developed, a dummy form would suffice.
- For additional details, please follow this link to the [wireframes thread](https://sef.discourse.group/t/designing-wireframes-for-scholarx-platform/148).
image

### PR DESCRIPTION
**Pull request by anjula-sack**
## Purpose

- The purpose of this PR is to fix #9 

## Goals

-  Creating manage mentee view

## Approach

- Create the mentee mentor card

## Screenshots

image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { Avatar, Button, Card, Typography } from 'antd';
+import { UserOutlined, CheckOutlined, CloseOutlined } from '@ant-design/icons';
+import styles from './styles.css';
+
+type Prop = {
+  name: string,
+};
+
+const { Title } = Typography;
+const MentorMenteeCard = ({ name }: Prop) => {
```
Comment (jayasanka-sack) [inline src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx]: Shall we go with a generic name like PersonCard?

File: src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (jayasanka-sack) [inline src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx]: the file name should be index.tsx

**Review by jayasanka-sack — COMMENTED** [review]
This is awesome @anjula-sack !

By the way, I'm worried about the following,
- There might be more than 10 cards in some scenarios. In those cases, this should be capable of handling paginations. Do you have any suggestions for it?  
- The actions can be changed according to the place. Maybe we don't have to worry about it at this level. But in those cases, the component should be extensible.

File: src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx (hunk 2 above)
Comment (anjula-sack) [reply src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx]: ah yes

File: src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx (hunk 1 above)
Comment (anjula-sack) [reply src/scenes/Dashboard/components/MentorMenteeCard/MentorMenteeCard.tsx]: yeah It's better

**Comment by Gravewalker666** [comment]
> * The actions can be changed according to the place. Maybe we don't have to worry about it at this level. But in those cases, the component should be extensible.

I think if we make this to render conditionally on a passed in prop we could achieve that flexibility. So in order to do that,
* First go through the wireframes and note all the possible renderings of the card
* Then define a prop as `type` or something to always have a value corresponding to a possible design   
* Use conditional rendering to render the action buttons
What do you think @jayasanka-sack ?

**Comment by jayasanka-sack** [comment]
> > * The actions can be changed according to the place. Maybe we don't have to worry about it at this level. But in those cases, the component should be extensible.
> 
> I think if we make this to render conditionally on a passed in prop we could achieve that flexibility. So in order to do that,
> 
> * First go through the wireframes and note all the possible renderings of the card
> * Then define a prop as `type` or something to always have a value corresponding to a possible design
> * Use conditional rendering to render the action buttons
>   What do you think @jayasanka-sack ?

My suggestion is to make component independent as much as possible. i.e. The component doesn't care where it is. It doesn't have much logic to adapt to the parent. The parent will decide how the component should behave. 

I'll paraphrase the issues again. 
   
1. The actions and the behaviors of actions can be varied depends on the scene 
  The actions, the logic behind the actions, API URLs can be varied depends on the scene. 
   ex: 
    **example scenario A:** if the component mounted on the "approve mentor" scene, there will be two buttons named approve and reject. If the mentor clicks on the approve button, it will call an API and do some actions. 
     **example scenario B:** if the component mounted on "apply for a mentor" view, there will be a single button named "apply". When a user clicks on the apply button, now it will display a model with a question list and an input box to submit answers. If the user has already applied, the button will be renamed as "edit application" and need to send another API call to fetch auto-fill the model with previously entered values.
     
1. The component may have additional details in some scenes 
    There might be some additional details in some scenes. Ex. Apply for a mentor may have a short description of the mentor. 

1. Some scenes may need pagination and some don't

As you can see, it's going to be complex with requirements and there's a high chance to be more complex with future requirements.  If we try to handle all of the above in a single component, then we have to go through all of the possibilities and come up with a too generic component with hell a lot of logic in it because it is not independent and highly depends on the parent or where it mounts. Since there are many logics in it, there's a high chance to be a "hard to understand", and a not well-performing component. 

As @anjula-sack and I discussed yesterday, one of the possible less complex implementations to address the mentioned issues is using the [antd's list component](https://ant.design/components/list) without introducing a new. 
1. The pagination is built-in with the component
1. The actions can be passed to the component as an array of react nodes. The parent will decide the actions. 
1. The appearance can be highly customized within the parent. 

 WDYT?

**Comment by Gravewalker666** [comment]
Yeah, totally agree with you, I think it's better if we can use ant-design list component for it is much flexible than we can ever get. We could add CSS styles to make it look like our initial design if we need to.

**Comment by jayasanka-sack** [comment]
Closing because the task has been completed with #17

**Pull request closed without merging**